WebAPI/CurrencyConverter/Data/ExchangeRateRepository.cs

```diff
@@ -0,0 +1,54 @@
+﻿using CurrencyConverter.Interfaces;
+using Model;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace CurrencyConverter.Data
+{
+    public class ExchangeRateRepository : IExchangeRateRepository
+    {
+        protected readonly ExchangeRateContext _context;
+        public ExchangeRateRepository(ExchangeRateContext context)
+        {
+            _context = context;
+        }
+        public void Add(ExchangeRate exchangeRate)
+        {
+            _context.Set<ExchangeRate>().Add(exchangeRate);
+        }
+        public void AddRange(IEnumerable<ExchangeRate> exchangeRates)
+        {
+            _context.Set<ExchangeRate>().AddRange(exchangeRates);
+        }
+        public IEnumerable<ExchangeRate> Find(Expression<Func<ExchangeRate, bool>> expression)
+        {
+            return _context.Set<ExchangeRate>().Where(expression);
+        }
+        public IEnumerable<ExchangeRate> GetAll()
+        {
+            return _context.Set<ExchangeRate>().ToList();
+        }
+        public ExchangeRate Get(int id)
+        {
+            return _context.Set<ExchangeRate>().Find(id);
+        }
+        public ExchangeRate GetByName(string currencyName)
+        {
+            return _context.Set<ExchangeRate>().Find(currencyName);
```

Comment:
Probably it will throw exception (I'm not sure), because primary key time is int, but we call Find method with string parameter.
May be we can use SingleOrDefault or FirstOrDefault?